### DISCUSSION TITLE
Bug: Incorrect file extension added

### DISCUSSION THREAD

**Discussion by reefhound** [mifi/lossless-cut#1353] [General]
Ver 3.47.1

Given filename GH011086.MP4
If I make a cut I get GH011086-00.00.01.001-00.00.20.020.MP4

This is perfect.

Given filename P8190048.MOV.MP4
If I make a cut I get P8190048.MOV-00.00.02.035-00.00.12.546.mov

The extension was mp4 and should remain mp4. The reason the filename is as above is because it was an MOV that was converted to MP4 and it was easiest to simply append the new extension and it tells me at a glance that it was converted to mp4 versus natively recorded in mp4.

**Comment by reefhound** [comment]
Just to add, it isn't the MOV extension causing the issue. I removed MOV from the filename. It appears that it is seeing something in the metadata to make it think it's an MOV file and not going by the MP4 extension.

Given filename P8190048.MP4
If I make a cut I get P8190048-00.00.02.035-00.00.12.546.mov

Exiftool shows these tags in the original:
MIME Type                       : video/mp4
Compatible Brands               : isom, iso2, avc1, mp41
Major Brand                     : MP4 Base Media v1 [IS0 14496-12:2003]
Minor Version                   : 0.2.0

I also have the setting "Try to automatically convert to supported format when opening unsupported file?" unchecked.

**Reply by mifi** [reply]
It's a known problem that some MP4 files LosslessCut (ffmpeg) is not able to export them as MP4 and needs to use MOV instead.

**Comment by reefhound** [comment]
That makes no sense as I'm currently fixing them in a batch file by using ffmpeg so I know ffmpeg can write these files correctly. Are you using the correct MajorBrand value?

%ffpath%\ffmpeg -i %clip% -metadata -majorbrand="MP4 v1 [ISO 14496-1:ch13]" -y -c copy %temp%
del %clip% /q
set clip=%~n1.MP4
RENAME %temp% %clip%

**Reply by mifi** [reply]
So if you remove -metadata .. from your ffmpeg command which error are you getting?

Could you provide extended codec information about the source file (help->report)?

**Comment by reefhound** [comment]
This isn't a show stopper for me as when I complete a folder I just paste the folder onto my batch file which loops through mov extensions and runs the above ffmpeg command. It would just be nice to eliminate an extra step.

**Comment by reefhound** [comment]
I'm not getting any error messages. After deeper testing, I've concluded that my issue is how I am using ffmpeg to convert MOV to MP4. Seems I am not re-encoding just changing the container. Apparently Lossless is using the codec not the container to determine file extension. I can easily have ffmpeg re-encode the libx264 or libx265 and solve teh file extension issue but it sort of defeats the whole point of "lossless".

It would be nice to have an option to preserve the container type.

**Reply by mifi** [reply]
I'm trying to get to the bottom of this. It seems like ffprobe `format_name` [does not contain the exact format but rather a list `mov,mp4,m4a,3gp,3g2,mj2`](https://superuser.com/questions/1586319/ffmpeg-does-not-change-container-from-mov-to-mp4) of the file. So we need to find some way to know which of these muxers (formats) to use when outputting. I'm going to rework this logic a bit.

On the other side, as can be seen in #948 - some codecs (pcm audio) need MOV when exporting, even if input is mp4. So losslesscut will default to `mov` output format if it detects those. Not sure if that's what's happening for you, and I don't know the details of your files (if you can include a report from the help menu it would be helpful), but you can try to select mp4 output format instead of `mov` and see if that helps.
